src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,61 @@
+import { nanoid } from "nanoid";
+import PropTypes from "prop-types";
+import React, { ReactElement, useRef } from "react";
+
+import "./modal.scss";
+
+type Props = {
+  buttons?: ReactElement[];
+  children?: ReactElement;
+  close?: () => void;
```

Comment:
For optional props I tend to add `| null` so that if props are defined by other code (e.g. the result of a function) then they may be `null` and then we have to define the prop as `close={onClose || undefined}` to satisfy the type.